R/autoxgboost.R

```diff
@@ -104,11 +106,21 @@ autoxgboost = function(task, measure = NULL, control = NULL, par.set = NULL, max
   }
 
   if (has.cat.feats) {
-    if (factor.encoder == "impact") {
-      base.learner = makeImpactFeaturesWrapper(base.learner)
-    } else {
-      task = createDummyFeatures(task)
+    d = getTaskData(task)
```

Comment:
you can set target.extra=TRUE, and select the first element, than you don't need the second line